src/timestamp.rs

```diff
@@ -0,0 +1,138 @@
+use std::{ops, fmt};
+
+/// The number of milliseconds, monotonically increasing since an
+/// arbitrary moment in time, such as system startup.
+#[derive(Debug, Default, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct Timestamp(pub u64);
+
+impl Timestamp {
+    /// Add a timestamp value or integer to the current timestamp
+    /// value. If the value is an integer the integer is assumed
+    /// to be a number of milliseconds.
+    pub fn advance<T: Into<Timestamp>>(&self, other: T) -> Timestamp {
```

Comment:
Ummmm, no. The *whole point* of having newtypes here is so that *timestamps* (absolute) and *intervals* (relative) cannot be accidentally mixed (which happened several times).

Reply:
Look at how libstd implements this. We need more or less the same thing, but simpler (it only has to handle milliseconds, and it doesn't have to hide its internals quite as ardently).

Reply:
> Ummmm, no. The whole point of having newtypes here is so that timestamps (absolute) and intervals (relative)

So do we want both a relative and absolute time type?

Reply:
like libstd's `Duration` and `Timespec`

Reply:
Exactly.